pkg/controllers/managementuser/resourcequota/resource_quota_common.go

```diff
@@ -217,7 +217,7 @@ func getProjectContainerDefaultLimit(ns *corev1.Namespace, projectLister v3.Proj
 func getNamespaceResourceQuotaLimit(ns *corev1.Namespace) (*v32.ResourceQuotaLimit, error) {
 	value := getNamespaceResourceQuota(ns)
 	if value == "" {
-		return nil, nil
+		return &v32.ResourceQuotaLimit{}, nil
```

Comment:
I realize that you are trying to avoid a `nil, nil` return here, but I am a bit uncomfortable with a blank limit being the default for this function. Especially because the newly renamed `deriveRequestedResourceQuota` function does a nil check for this value.

I think that the the `completeQuota` function looks like it can handle a blank default value (it will just not replace anything from the project), but the case of a nil project limit and a nil namespace limit (where we don't want any limits/quotas set) is now going to be different (you will return a zero quota limit and not a nil) for `deriveRequestedResourceQuota`. I would double check that there is no difference between no ResourceQuotaLiimit and a `ResourceQuotaLimit{}`

Reply:
You're right, at this point there is no need to have this non-nil pointer, because the case for it to be explicitly non-nil is moot, since we thoroughly check the value of `requested` and `default` and their combinations.